code/datums/traits.dm

```diff
@@ -141,7 +141,20 @@
 			other.addTrait(id, traits[id])
 
 	proc/addTrait(id, datum/trait/trait_instance=null)
-		if(!(id in traits))
+		// Blacklist for harmful afterlife traits that shouldn't be added
+		var/blacklist_afterlife = list(
+			"deaf",
+			"shortsighted",
+			"blind",
+			"mildly_mutated",
+			"addict",
+			"randomallergy",
+			"medicalallergy",
+			"allergic",
+			"clutz",
+			"leftfeet"
+		)
```

Comment:
Make this a `var/afterlife_blacklisted = FALSE` on the trait instead and override it to `TRUE` on the relevant ones. 
Then just check `T.afterlife_bkaclisted` instead of checking this list.

Reply:
Done!